### PR TITLE
Support 'report-sample' keyword-source

### LINKED ISSUE #183: Support `report-sample` keyword-source
https://w3c.github.io/webappsec-csp/#grammardef-report-sample

### PR DESCRIPTION
**Pull request by shekyan**
fixes #183

### CODE REVIEW

File: src/test/java/com/shapesecurity/salvation/ParserTest.java [hunk 1]
```diff
@@ -1050,4 +1050,34 @@
...
+        p = parseWithNotices("script-src 'report-sample' 'report-sample'", notices);
+        assertEquals(1, p.getDirectives().size());
+        assertEquals("script-src 'report-sample'", p.getDirectiveByType(ScriptSrcDirective.class).show());
+        assertEquals(0, notices.size());
+
+        notices.clear();
+        p = parseWithNotices("style-src 'report-sample' 'unsafe-hashed-attributes' 'report-sample' 'nonce-123'", notices);
+        assertEquals(1, p.getDirectives().size());
+        assertEquals(2, notices.size());
+        assertEquals("Invalid base64-value (should be multiple of 4 bytes: 3).", notices.get(0).message);
```
Comment (michaelficarra) [inline src/test/java/com/shapesecurity/salvation/ParserTest.java]: You have two notices and the one you're asserting on is unrelated to report-sample. Just fix the nonce and assert on the remaining notice.

File: src/test/java/com/shapesecurity/salvation/ParserTest.java [hunk 2]
```diff
@@ -1050,4 +1050,34 @@
...
+        ArrayList<Notice> notices = new ArrayList<>();
+        p = parseWithNotices("default-src 'report-sample'", notices);
+        assertEquals(1, p.getDirectives().size());
+        assertEquals(0, notices.size());
+
+        notices.clear();
+        p = parseWithNotices("script-src 'report-sample' 'report-sample'", notices);
+        assertEquals(1, p.getDirectives().size());
+        assertEquals("script-src 'report-sample'", p.getDirectiveByType(ScriptSrcDirective.class).show());
+        assertEquals(0, notices.size());
```
Comment (michaelficarra) [inline src/test/java/com/shapesecurity/salvation/ParserTest.java]: Two report-samples, but no notices?

Comment (shekyan) [inline src/test/java/com/shapesecurity/salvation/ParserTest.java]: Will be addressed in #185

File: src/test/java/com/shapesecurity/salvation/ParserTest.java (hunk 1 above)
Comment (shekyan) [inline src/test/java/com/shapesecurity/salvation/ParserTest.java]: Both notices are unrelated to `report-sample`, removed the test

File: src/main/java/com/shapesecurity/salvation/Parser.java [hunk 3]
```diff
@@ -391,6 +391,8 @@ private void enforceMissingDirectiveValue(@Nonnull Token directiveNameToken) thr
                     seenStates.add(SeenStates.SEEN_HOST_OR_SCHEME_SOURCE);
                 } else if (se == KeywordSource.UnsafeHashedAttributes) {
                     seenStates.add(SeenStates.SEEN_UNSAFE_HASHED_ATTR);
+                } else if (se == KeywordSource.ReportSample) {
+                    seenStates.add(SeenStates.SEEN_REPORT_SAMPLE);
```
Comment (michaelficarra) [inline src/main/java/com/shapesecurity/salvation/Parser.java]: It seems silly to add this if it's not going to be read, but whatever.

**Pull request merged**